src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -249,13 +249,18 @@
     </xsd:complexType>
 
     <xsd:complexType name="oauth2-client-credentials-type">
+        <xsd:choice minOccurs="0" maxOccurs="1">
+            <xsd:element name="credential-store-reference" type="credential-store-reference-type"/>
+        </xsd:choice>
         <xsd:attribute name="client-id" type="xsd:string" use="required" />
-        <xsd:attribute name="client-secret" type="xsd:string" use="required" />
+        <xsd:attribute name="client-secret" type="xsd:string" use="optional" />
```

Comment:
I think there is the same problem as for **resource-owner-credentials-type** - when **credential-store-reference-type** is available, there is no need to keep **client-secret** attribute.